cpu/nrf52/periph/i2c.c

```diff
@@ -167,38 +173,31 @@ void i2c_release(i2c_t dev)
 int i2c_write_regs(i2c_t dev, uint16_t addr, uint16_t reg,
                    const void *data, size_t len, uint8_t flags)
 {
-    assert((dev < I2C_NUMOF) && data && (len > 0) && (len < 255));
+    assert((dev < I2C_NUMOF) && data && (len > 0) && (len < 253));
 
     if (flags & (I2C_NOSTART | I2C_ADDR10)) {
         return -EOPNOTSUPP;
     }
 
-    /* the nrf52's TWI device does not support to do two consecutive transfers
-     * without a repeated start condition in between. So we have to put all data
-     * to be transferred into a temporary buffer
-     *
-     * CAUTION: this might become critical when transferring large blocks of
-     *          data as the temporary buffer is allocated on the stack... */
+    /* Resets the TX buffer */
+    memset(tx_buf, 0, sizeof(tx_buf));
```

Comment:
That should not be needed, we already overwrite it with `memcpy` and do not transfer more bytes than written to it.